src/utstring.h

```diff
@@ -121,6 +121,24 @@ do {                                                             \
 
 #define utstring_body(s) ((s)->d)
 
+#define utstring_adopt(s,b,l)                              \
+do {                                                       \
+  if (s) {                                                 \
+    (s)->d = (b);                                          \
+    (s)->n = (l);                                          \
+    (s)->i = (l);                                          \
```

Comment:
@troydhanson, in the codebases where you've needed this function in the past, would this have sufficed? Or should it be `utstring_adopt(s,d,i,n)` for cases where the initial capacity is intentionally greater than the initial length?
I guess since C's `realloc` and `free` don't need to know the original capacity, this version is safe; but if someone later filed a feature request to make utstring work with `ptr = sizedmalloc(sz)` and `sizedfree(ptr,sz)` the way uthash currently does, it would be hard to retrofit that into this "length-only" model.

Reply:
@tbeu: Sorry! I think what happened here was that the feedback I was awaiting on 2016-12-15 *did* arrive, in the form of Troy's comment on 2016-12-16. And then I never explicitly said "@tbeu please address this," but I also never really bothered (AFAIR) to figure out whether I even agreed with what Troy said, either.

I suggest that at this point, with many outdated review comments and some pieces of this patch merged already, the best thing to do would be to close this PR and (if you still want the feature merged) open a fresh PR.

*Do* you have any thoughts on the desirability of `utstring_adopt(s, buf, len)` versus `utstring_adopt(s, ntbs)` versus `utstring_adopt(s, buf, len, cap)`?  I think that (at the moment) I agree with Troy that `utstring_adopt(s, buf, len)` is the worst of both worlds, because it would let you create a non-null-terminated utstring.